webui/react/src/shared/types.ts

```diff
@@ -33,9 +33,22 @@ export interface DetApi<Input, DetOutput, Output> extends ApiBase {
   stubbedResponse?: DetOutput;
 }
 
+/**
+ * @description helper to organize storing api response data.
+ */
 export interface ApiState<T> {
   data?: T;
+  /**
+   * error, if any, with the last state update.
+   * this should be cleared on the next successful update.
+  */
   error?: Error;
+  /**
+   * indicates whether the state has been fetched at least once or not.
+   * should always be initialized to false.
+   */
+  hasBeenInitialized: boolean;
```

Comment:
optional? else update 

https://github.com/determined-ai/determined/blob/db32a6434dc9e7e54e53b7f720021ff865dc133c/webui/react/src/pages/TrialDetails.tsx#L52

Reply:
could leave it as optional 🤔 . yeah let me do that instead